File: WMS.EntityCore/Trans_oc/clsBeTrans_oc_det_lote.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.EntityCore.Producto;

namespace WMSWebAPI.Be
{
    public class clsBeTrans_oc_det_lote : ICloneable
    {
        [Column("IdOrdenCompraEnc")]
        [DisplayName("IdOrdenCompraEnc")]
        public int IdOrdenCompraEnc { get; set; } = 0;

        [Column("IdOrdenCompraDet")]
        [DisplayName("IdOrdenCompraDet")]
        public int IdOrdenCompraDet { get; set; } = 0;

        [Column("IdOrdenCompraDetLote")]
        [DisplayName("IdOrdenCompraDetLote")]
        public int IdOrdenCompraDetLote { get; set; } = 0;

        [Column("IdProductoBodega")]
        [DisplayName("IdProductoBodega")]
        public int IdProductoBodega { get; set; } = 0;

        [Column("no_linea")]
        [DisplayName("no_linea")]
        public int No_linea { get; set; } = 0;

        [Column("codigo_producto")]
        [DisplayName("codigo_producto")]
        public string Codigo_producto { get; set; } = "";

        [Column("cantidad")]
        [DisplayName("cantidad")]
        public double Cantidad { get; set; } = 0;

        [Column("cantidad_recibida")]
        [DisplayName("cantidad_recibida")]
        public double Cantidad_recibida { get; set; } = 0;

        [Column("lote")]
        [DisplayName("lote")]
        public string Lote { get; set; } = "";

        [Column("fecha_vence")]
        [DisplayName("fecha_vence")]
        public DateTime Fecha_vence { get; set; } = DateTime.Now;

        [Column("lic_plate")]
        [DisplayName("lic_plate")]
        public string Lic_plate { get; set; } = "";

        [Column("Ubicacion")]
        [DisplayName("Ubicacion")]
        public string Ubicacion { get; set; } = "";

        [Column("IdPresentacion")]
        [DisplayName("IdPresentacion")]
        public int IdPresentacion { get; set; } = 0;

        [Column("IdUnidadMedidaBasica")]
        [DisplayName("IdUnidadMedidaBasica")]
        public int IdUnidadMedidaBasica { get; set; } = 0;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("reclasificar")]
        [DisplayName("reclasificar")]
        public bool Reclasificar { get; set; } = false;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("no_documento")]
        [DisplayName("no_documento")]
        public string No_documento { get; set; } = "";
        public clsBeProducto_presentacion Presentacion { get; set; } = new clsBeProducto_presentacion();
        public clsBeUnidad_medida UnidadMedida { get; set; } = new clsBeUnidad_medida();
        public int IdProductoTallaColor { get; set; } = 0;
        public string Talla { get; set; } = "";
        public string Color { get; set; } = "";

        public clsBeTrans_oc_det_lote() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}

File: TOMIMSV4/Entity/Transacciones/OrdenCompra/OC_Detalle_Lote/clsBeTrans_oc_det_lote.vb
Public Class clsBeTrans_oc_det_lote
    Implements ICloneable

    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property IdOrdenCompraDet() As Integer = 0
    Public Property IdOrdenCompraDetLote() As Integer = 0
    Public Property IdProductoBodega() As Integer = 0
    Public Property No_linea() As Integer = 0
    Public Property Codigo_producto() As String = ""
    Public Property Cantidad() As Double = 0.0
    Public Property Cantidad_recibida() As Double = 0.0
    Public Property Lote() As String = ""
    Public Property Lic_Plate As String = ""
    Public Property Fecha_vence() As Date = New Date(1900, 1, 1)
    Public Property Ubicacion As String = ""
    Public Property IdPresentacion() As Integer = 0
    Public Property IdUnidadMedidaBasica() As Integer = 0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now

    '#CKFK20220212 Se agregó este campo para cuando en la recepción de une devolución de venta no se reciba el mismo lote y se tenga que enviar el correcto al ERP
    Public Property Reclasificar As Boolean = False
    Public Property IsNew As Boolean = False
    Public Property Activo As Boolean = True
    Public Property No_Documento As String = ""
    Public Property Codigo_Sku As String = ""
    Public Property IdProductoTallaColor As Integer = 0
    Public Property Talla As String = ""
    Public Property Color As String = ""

    Public Peso_Licencia As Double = 0

    Sub New()
    End Sub

    Sub New(ByRef IdOrdenCompraEnc As Integer, ByVal IdOrdenCompraDet As Integer, ByVal IdOrdenCompraDetLote As Integer, ByVal IdProductoBodega As Integer, ByVal no_linea As Integer, ByVal codigo_producto As String, ByVal cantidad As Double, ByVal cantidad_recibida As Double, ByVal lote As String, ByVal fecha_vence As String)
        Me.IdOrdenCompraEnc = IdOrdenCompraEnc
        Me.IdOrdenCompraDet = IdOrdenCompraDet
        Me.IdOrdenCompraDetLote = IdOrdenCompraDetLote
        Me.IdProductoBodega = IdProductoBodega
        Me.No_linea = no_linea
        Me.Codigo_producto = codigo_producto
        Me.Cantidad = cantidad
        Me.Cantidad_recibida = cantidad_recibida
        Me.Lote = lote
        Me.Fecha_vence = fecha_vence
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class
